### PR TITLE
Bitmap brush

### PR DESCRIPTION
**Pull request by fsih**
### Resolves

Toward adding support for bitmap tools

### Proposed Changes

Adds the bitmap brush tool
bitbrush

### CODE REVIEW

File: src/helper/bitmap.js [hunk 1]
```diff
@@ -1,5 +1,85 @@
...
+    let err = dx - dy;
+    
+    callback(x1, y1);
+    while (x1 !== x2 || y1 !== y2) {
+        const e2 = err * 2;
+        if (e2 > -dy) {
+            err -= dy; x1 += sx;
+        }
+        if (e2 < dx) {
+            err += dx; y1 += sy;
```
Comment (paulkaplan) [inline src/helper/bitmap.js]: probably want only single statement per line!

File: src/helper/bitmap.js [hunk 2]
```diff
@@ -1,5 +1,85 @@
...
+        }
+        callback(x1, y1);
+    }
+};
+
+const fillEllipse = function (centerX, centerY, radiusX, radiusY, context) {
+    // Bresenham ellipse algorithm
+    centerX = ~~centerX;
```
Comment (paulkaplan) [inline src/helper/bitmap.js:33]: https://jsperf.com/tilde-vs-floor/11 I kinda doubt the `~~` is actually faster at this point, but I don't have a super strong opinion ¯\_(ツ)_/¯

File: src/helper/bitmap.js [hunk 3]
```diff
@@ -1,5 +1,85 @@
 import paper from '@scratch/paper';
 
+const line = function (point1, point2, callback) {
```
Comment (paulkaplan) [inline src/helper/bitmap.js]: Might want to rename this to something like `forEachLinePoint`, since this doesn't actually draw a line.

**Review by paulkaplan — DISMISSED** [review]
A few minor comments that I'll leave up to you. This looks and works great! 

Still having some issues with image getting displaced switching between bitmap and vector, but I know you already have that filed.

File: src/helper/bitmap.js (hunk 2 above)
Comment (fsih) [reply src/helper/bitmap.js:33]: Ok, I took out the comment that said it was faster :p

File: src/helper/bitmap.js (hunk 3 above)
Comment (fsih) [reply src/helper/bitmap.js]: Done

File: src/helper/bitmap.js (hunk 1 above)
Comment (fsih) [reply src/helper/bitmap.js]: Done

**Pull request merged**